clusterloader2/pkg/framework/client/objects.go

```diff
@@ -154,3 +157,17 @@ func GetObject(dynamicClient dynamic.Interface, gvk schema.GroupVersionKind, nam
 	}
 	return obj, nil
 }
+
+func createPatch(current, modified *unstructured.Unstructured) ([]byte, error) {
+	currentJson, err := current.MarshalJSON()
+	if err != nil {
+		return []byte{}, err
+	}
+	modifiedJson, err := modified.MarshalJSON()
+	if err != nil {
+		return []byte{}, err
+	}
+	preconditions := []mergepatch.PreconditionFunc{mergepatch.RequireKeyUnchanged("apiVersion"),
+		mergepatch.RequireKeyUnchanged("kind"), mergepatch.RequireMetadataKeyUnchanged("name")}
+	return jsonmergepatch.CreateThreeWayJSONMergePatch(nil, modifiedJson, currentJson, preconditions...)
```

Comment:
Are you copying this logic over from somewhere?

Reply:
Since you're passing the first field of a 3-way merge (which is supposed to be the original object) as nil, that seems to be as good as just a 2-way merge with the new one being applied to the current one. Can't you instead just use `strategicpatch.CreateTwoWayMergePatch`?

Reply:
Actually I'm a bit unsure if both of them are the same. IIUC using TwoWayMergePatch means that whenever there's a conflict and we need to choose, the new one will simply override the old one. So we don't detect for conflicts before applying. However, ThreeWayMergePatch will throw an error with a conflict so we can get and retry again - but for that we need to pass original object I think?

@liggitt Could you give a lead on what's the right approach to follow here?

Reply:
CreateTwoWayMergePatch requires actual object structure to be passed. Currently we only use Unstructured which is not supported by this function.
TODO added.

Reply:
Thanks for adding todo. Let's figure this out later if that's the right way.